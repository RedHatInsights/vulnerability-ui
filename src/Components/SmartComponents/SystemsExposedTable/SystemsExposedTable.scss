.affected-systems-toolbar {
    justify-content: flex-start;
    
    > div {
        margin-right: var(--pf-global--spacer--sm);
    }
}

.ins-composed-col {
    display: inline;
}

.affected-systems-filter-chips {
    order: 5;
    flex-basis: 100%;
}
