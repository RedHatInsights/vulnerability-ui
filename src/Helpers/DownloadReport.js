import { downloadFile } from '@redhat-cloud-services/frontend-components-utilities/files/helpers';

class DownloadReport  {
    constructor() {
        this.defaultParams = {
            page: 1,
            page_size: Number.MAX_SAFE_INTEGER,
            date: new Date().toISOString().replace(/[T:]/g, '-').split('.')[0] + '-utc'
        };

        this.params = {
            cve: ''
        };

        this.pages = ['systems-exposed', 'system-cves', 'cves', 'system-list'];
        this.formats = ['csv', 'json'];
    }

    filename = (page) => {
        switch (page) {
            case 'systems-exposed':
                return `vulnerability_systems-exposed-${this.params.cve}--${this.defaultParams.date}`;
            case 'system-list':
                return `vulnerability_systems--${this.defaultParams.date}`;
            default:
                return `vulnerability_cves--${this.defaultParams.date}`;

        }
    }

    checkTypes = (fetchData, page, format) => {
        if (typeof fetchData !== 'function') {
            throw `exec expects the 1st param to be funtion. Received ${typeof fetchData}`;
        }

        if (!this.pages.includes(page)) {
            throw `Page ${page} does not exist`;
        }

        if (!this.formats.includes(format)) {
            throw `Format ${format} is not supported`;
        }
    }

    /**
     * Execute and download the report
     *
     * @param {Function} fetchData - Function to be called in order to fetch data.
     * @param {Object || Array} params - Object to be sent as request params. If you need to pass extra argument, pass an array.
     * @param {String} format - Supported formats to download are CSV and JSON.
     * @param {String} page - Name of the page based on which the filename will be determined.
     * @returns {void}
     */
    exec = async (fetchData, params, format, page) => {
        this.checkTypes(fetchData, page, format);
        this.params.cve = Array.isArray(params) ? params[0] : '';

        try {
            let { payload } = Array.isArray(params)
                ? await fetchData(this.params.cve, {
                    ...params[1],
                    ...this.defaultParams,
                    data_format: format
                }) :
                await fetchData({
                    ...params,
                    ...this.defaultParams,
                    data_format: format
                });

            let { data: response } = await payload;
            let data = format === 'json' ? JSON.stringify(response) : response;
            // downaloads the files
            downloadFile(data, this.filename(page), format);
        } catch (error) {
            throw `${error}`;
        }

    }

}

export default new DownloadReport();
