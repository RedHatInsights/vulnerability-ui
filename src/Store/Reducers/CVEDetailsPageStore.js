import Immutable from 'seamless-immutable';
import * as ActionTypes from '../ActionTypes';

// Initial State
export const initialState = Immutable({
    affectedSystemsByCVE: {
        isLoading: true,
        openedRows: [],
        payload: {
            meta: {
                total_items: 0,
                page: 1,
                page_size: 20
            }
        }
    },
    parameters: {
        page: 1,
        page_size: 20,
        selectedHosts: [],
        status_id: undefined,
        sort: '-last_upload',
        show_advisories: true
    },
    cveDetails: {
        isLoading: true,
        payload: {},
        error: false
    }
});

// Reducer
export const CVEDetailsPageStore = (state = initialState, action) => {
    let newState;
    switch (action.type) {
        case ActionTypes.FETCH_CVE_DETAILS + '_REJECTED':
            newState = state.setIn(['cveDetails', 'error'], true);
            newState = newState.setIn(['cveDetails', 'isLoading'], false);
            return newState;
        case ActionTypes.FETCH_CVE_DETAILS + '_PENDING':
            newState = state.setIn(['cveDetails', 'isLoading'], true);
            return newState;

        case ActionTypes.FETCH_CVE_DETAILS + '_FULFILLED':
            newState = state.setIn(['cveDetails', 'payload'], action.payload);
            newState = newState.setIn(['cveDetails', 'isLoading'], false);
            return newState;

        case ActionTypes.FETCH_AFFECTED_SYSTEMS_BY_CVE + '_PENDING':
            newState = state.setIn(['affectedSystemsByCVE', 'isLoading'], true);
            return newState;

        case ActionTypes.FETCH_AFFECTED_SYSTEMS_BY_CVE + '_FULFILLED':
            newState = state.setIn(['affectedSystemsByCVE', 'payload'], action.payload);
            newState = newState.setIn(['affectedSystemsByCVE', 'isLoading'], false);
            return newState;
        case ActionTypes.CHANGE_EXPOSED_SYSTEMS_PARAMETERS:
            newState = state.setIn(['parameters'], {
                ...state.parameters,
                ...action.payload,
                page_size: action.payload.page_size || state.parameters.page_size
            });
            return newState;
        case ActionTypes.SET_GLOBAL_FILTER:
            newState = state.setIn(['parameters'], {
                ...state.parameters,
                ...action.payload
            });
            return newState;
        case ActionTypes.CLEAR_CVE_STORE:
            return initialState;
        default:
            return state;
    }
};

