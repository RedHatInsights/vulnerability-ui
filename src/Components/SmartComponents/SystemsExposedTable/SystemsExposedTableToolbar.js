
import React, { Fragment } from 'react';
import propTypes from 'prop-types';
import { useIntl } from 'react-intl';
import { PrimaryToolbar } from '@redhat-cloud-services/frontend-components/PrimaryToolbar';
import Remediation from '../Remediation/Remediation';
import messages from '../../../Messages';
import securityRuleFilter from '../../PresentationalComponents/Filters/PrimaryToolbarFilters/SecurityRuleFilter';
import statusFilter from '../../PresentationalComponents/Filters/PrimaryToolbarFilters/StatusFilter';
import useOsVersionFilter from '../../PresentationalComponents/Filters/PrimaryToolbarFilters/OsVersionFilter';
import { exportConfig, buildActiveFilters, removeFilters } from '../../../Helpers/TableToolbarHelper';
import { dataShape } from '../../../Helpers/MiscHelper';
import { useBulkSelect } from '../../../Helpers/Hooks';
import useSearchFilter from '../../PresentationalComponents/Filters/PrimaryToolbarFilters/SearchFilter';
import {
    fetchAffectedSystemsIdsByCve
} from '../../../Store/Actions/Actions';
import {
    RULE_ABSENCE_OPTIONS, ANSIBLE_REMEDIATION
} from '../../../Helpers/constants';
import remediationFilter from '../../PresentationalComponents/Filters/PrimaryToolbarFilters/RemediationFilter';

export const SystemsExposedTableToolbar = ({
    rawData,
    cveName,
    cveStatusDetails,
    selectedRows,
    selectedRowsCount,
    selectedRowsRawData,
    filterRuleValues,
    parameters,
    methods,
    canEditPairStatus,
    canExport,
    children
}) => {
    const intl = useIntl();
    const { apply, showStatusModal, handleSelect, downloadReport, setColumnModalOpen } = methods;
    const { isLoaded, meta } = rawData;

    const remediableSystems = selectedRowsRawData.filter(system => system.remediation === ANSIBLE_REMEDIATION);

    const kebabOptions = ['',
        ...canEditPairStatus ? [{
            label: intl.formatMessage(messages.editStatus),
            onClick: () => showStatusModal([cveStatusDetails], selectedRowsRawData),
            props: { isDisabled: !selectedRowsCount }
        }] : [],
        {
            label: intl.formatMessage(messages.columnManagementModalTitle),
            onClick: () => setColumnModalOpen(true)
        }
    ];

    const searchFilter = useSearchFilter(
        'filter',
        messages.systemsSearchName,
        messages.searchFilterByName,
        parameters.filter,
        apply
    );

    const advisoryFilter = useSearchFilter(
        'advisory',
        messages.advisory,
        messages.searchFilterByAdvisory,
        parameters.advisory,
        apply
    );

    const osVersionFilter = useOsVersionFilter(
        parameters.rhel_version,
        apply
    );

    const bulkSelectProps = useBulkSelect({
        rawData,
        selectedRows,
        selectedRowsCount,
        handleSelect,
        fetchResource: ops => fetchAffectedSystemsIdsByCve({ id: cveName, ...parameters, ...ops })
    });

    return <Fragment>
        <PrimaryToolbar
            className="vuln-systems-primary-toolbar"
            exportConfig={canExport && {
                isDisabled: meta.totalItems === 0,
                ouiaId: 'export',
                ...exportConfig({ downloadReport })
            }}
            dedicatedAction={(isLoaded &&
                <Remediation
                    manyRules
                    isDisabled={remediableSystems.length === 0}
                    systems={remediableSystems}
                    cves={{ id: cveName, rules: filterRuleValues }}
                />
            )}
            actionsConfig={{
                actions: kebabOptions,
                dropdownProps: { ouiaId: 'toolbar-actions' }
            }}
            activeFiltersConfig={{
                filters: buildActiveFilters({ ...parameters }, filterRuleValues),
                onDelete: (_, chips) => removeFilters(chips, apply),
                deleteTitle: intl.formatMessage(messages.resetFilters)
            }}
            bulkSelect={bulkSelectProps}
            filterConfig={{
                items: [
                    searchFilter,
                    securityRuleFilter(
                        apply,
                        parameters,
                        filterRuleValues,
                        {
                            isDynamic: true,
                            dropdownItems: RULE_ABSENCE_OPTIONS
                        }
                    ),
                    statusFilter(apply, parameters),
                    advisoryFilter,
                    osVersionFilter,
                    remediationFilter(apply, parameters)
                ]
            }}
        />
        {children}
    </Fragment>;
};

SystemsExposedTableToolbar.propTypes = {
    rawData: dataShape,
    parameters: propTypes.object,
    cveName: propTypes.string,
    cveStatusDetails: propTypes.object,
    filterRuleValues: propTypes.object,
    selectedRows: propTypes.array,
    selectedRowsCount: propTypes.number,
    selectedRowsRawData: propTypes.array,
    children: propTypes.node,
    methods: propTypes.shape({
        apply: propTypes.func,
        handleSelect: propTypes.func,
        showStatusModal: propTypes.func,
        downloadReport: propTypes.func,
        setColumnModalOpen: propTypes.func
    }),
    canEditPairStatus: propTypes.bool,
    canExport: propTypes.bool
};
