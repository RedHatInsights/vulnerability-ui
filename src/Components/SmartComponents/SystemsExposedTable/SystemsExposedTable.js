import * as reactCore from '@patternfly/react-core';
import {
    Dropdown,
    DropdownItem,
    KebabToggle,
    Text,
    TextContent,
    TextVariants,
    ToolbarGroup,
    ToolbarItem
} from '@patternfly/react-core';
import * as reactIcons from '@patternfly/react-icons';
import * as pfReactTable from '@patternfly/react-table';
import { FilterDropdown, SimpleTableFilter } from '@redhat-cloud-services/frontend-components';
import { StatusDropdown } from '@redhat-cloud-services/frontend-components-inventory-vulnerabilities';
import { addNotification } from '@redhat-cloud-services/frontend-components-notifications';
import RemediationButton from '@redhat-cloud-services/frontend-components-remediations/RemediationButton';
import { downloadFile } from '@redhat-cloud-services/frontend-components-utilities/files/helpers';
import debounce from 'lodash/debounce';
import omit from 'lodash/omit';
import propTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import * as reactRouterDom from 'react-router-dom';
import { withRouter } from 'react-router-dom';
import { dispatchAction } from '../../../Helpers/Dispatcher';
import { dataShape } from '../../../Helpers/MiscHelper';
import { fetchAffectedSystemsByCVE, fetchSystemCveStatusList } from '../../../Store/Actions/Actions';
import { inventoryEntitiesReducer } from '../../../Store/Reducers/InventoryEntitiesReducer';
import { sExposedSystemsTable } from '../../../Store/Selectors/CVESelectors';
import ReducerRegistry from '../../../Utilities/ReducerRegistry';
import './SystemsExposedTable.scss';
/*eslint camelcase: 0*/

class SystemsExposedTable extends React.Component {
    static propTypes = {
        cve: propTypes.string,
        fetchData: propTypes.func,
        affectedSystems: dataShape,
        entities: propTypes.object,
        statusList: propTypes.object,
        fetchStatusList: propTypes.func
    };

    constructor(props) {
        super(props);

        this.state = {
            InventoryCmp: () => <div>Loading...</div>,
            removeListener: () => {},
            filter: '',
            columns: [
                {
                    key: 'display_name',
                    title: 'Name',
                    composed: ['facts.os_release', 'display_name'],
                    props: {
                        width: 40
                    }
                },
                {
                    key: 'status',
                    title: 'Status',
                    props: {
                        width: 30
                    }
                },
                {
                    key: 'updated',
                    title: 'Last Seen',
                    isTime: true,
                    props: {
                        width: 30
                    }
                }
            ],
            isKebabOpen: false
        };
    }

    componentDidMount() {
        this.apply();
        this.props.fetchStatusList();
        StatusDropdown.setCallback(this.dataReload);
        this.fetchInventory();
    }

    addFilter = (param, value, type) => {
        let newFilter;
        if (type === 'radio' || param === 'show_all') {
            newFilter = { [param]: value };
        } else {
            newFilter = this.state[param] ? { [param]: `${this.state[param]},${value}` } : { [param]: value };
        }

        this.setState({ ...this.state, ...newFilter, page: 1 }, this.apply);
    };

    removeFilter = (key, value) => {
        const newFilter = {
            [key]: this.state[key]
            .split(',')
            .filter(item => item !== value)
            .join(',')
        };

        if (newFilter.length !== 0) {
            this.setState({ ...this.state, ...newFilter, page: 1 }, this.apply);
        } else {
            const filter = { ...this.state, [key]: undefined };
            this.setState({ ...this.state, ...filter, page: 1 }, this.apply);
        }
    };

    getSelectedSystems = () => {
        let systems =
            (this.props.entities &&
                this.props.entities.loaded &&
                this.props.entities.rows.filter(system => system.selected).map(system => system.id)) ||
            0;
        return systems;
    };

    dataProvider = () => {
        return {
            issues: [
                {
                    id: `vulnerabilities:${this.props.cve}`,
                    description: this.props.cve
                }
            ],
            systems: this.getSelectedSystems()
        };
    };

    dataReload = () => {
        const { meta } = this.props.affectedSystems;
        this.setState({ ...this.state, page: meta.page, page_size: meta.page_size, filter: this.state.filter }, this.apply);
    };

    apply = () => {
        const ignoredProps = ['isKebabOpen', 'columns'];
        const params = omit(this.state, ignoredProps);

        this.props.fetchData(this.props.cve, { ...params });
    };

    downloadReport = format => {
        const { payload } =
            // eslint-disable-next-line camelcase
            fetchAffectedSystemsByCVE(this.props.cve, {
                ...this.state,
                data_format: format,
                page: 1,
                page_size: Number.MAX_SAFE_INTEGER
            });
        payload &&
            payload.then(({ data: response }) => {
                const data = format === 'json' ? JSON.stringify(response) : response;
                return downloadFile(data, `systems_affected-${this.props.cve}-${new Date().toISOString()}`, format);
            });
    };

    async fetchInventory() {
        const { inventoryConnector, mergeWithEntities, mergeWithDetail } = await insights.loadInventory({
            react: React,
            reactRouterDom,
            reactCore,
            reactIcons,
            pfReactTable
        });
        ReducerRegistry.register({
            ...mergeWithEntities(inventoryEntitiesReducer(this.state.columns)),
            ...mergeWithDetail()
        });

        this.setState({
            InventoryCmp: inventoryConnector().InventoryTable
        });
    }

    handleFilterChange = debounce(value => {
        this.setState({ ...this.state, filter: value }, this.apply);
    }, 400);

    handleRemediationCreated = result => {
        dispatchAction(addNotification(result.getNotification()));
    };

    handleKebabToggle = isKebabOpen => {
        this.setState({
            isKebabOpen
        });
    };

    handleKebabSelect = () => {
        this.setState({
            isKebabOpen: !this.state.isKebabOpen
        });
    };

    handleRefresh = ({ page, per_page: perPage }) => {
        const { meta } = this.props.affectedSystems;
        if (meta.page !== page || meta.page_size !== perPage) {
            this.setState(
                {
                    ...this.state,
                    page: meta.page_size !== perPage ? 1 : page,
                    page_size: perPage,
                    filter: this.state.filter
                },
                this.apply
            );
        }
    };

    render() {
        const { InventoryCmp } = this.state;
        const { affectedSystems, statusList } = this.props;
        const { data, meta } = affectedSystems;
        const filtersStatusList = statusList &&
            statusList.payload && {
            type: 'checkbox',
            title: 'Status',
            urlParam: 'status_id',
            values: statusList.payload.data.map(item => ({ label: item.name, value: String(item.id) }))
        };
        return (
            <React.Fragment>
                <TextContent>
                    <Text component={TextVariants.h2}>Affected systems</Text>
                </TextContent>
                <InventoryCmp
                    hasCheckbox={data.length !== 0}
                    page={meta.page}
                    total={meta.total_items}
                    perPage={meta.page_size}
                    onRefresh={this.handleRefresh}
                    items={data.map(item => ({
                        id: item.attributes.inventory_id,
                        ...item.attributes
                    }))}
                >
                    <ToolbarGroup className={'affected-systems-toolbar'}>
                        <ToolbarItem>
                            <SimpleTableFilter
                                onFilterChange={this.handleFilterChange}
                                buttonTitle={null}
                                placeholder="Find a system by UUID"
                            />
                        </ToolbarItem>
                        <ToolbarItem>
                            {filtersStatusList && (
                                <FilterDropdown
                                    filterCategories={[filtersStatusList]}
                                    addFilter={this.addFilter}
                                    removeFilter={this.removeFilter}
                                    filters={this.state}
                                />
                            )}
                        </ToolbarItem>
                        <ToolbarItem>
                            <RemediationButton
                                dataProvider={this.dataProvider}
                                isDisabled={!this.getSelectedSystems().length}
                                onRemediationCreated={this.handleRemediationCreated}
                            />
                        </ToolbarItem>
                        <ToolbarItem>
                            <Dropdown
                                onSelect={this.handleKebabSelect}
                                toggle={<KebabToggle onToggle={this.handleKebabToggle} />}
                                isOpen={this.state.isKebabOpen}
                                isPlain
                                dropdownItems={[
                                    <DropdownItem key="json" component="button" onClick={() => this.downloadReport('json')}>
                                        Export as JSON
                                    </DropdownItem>,
                                    <DropdownItem key="csv" component="button" onClick={() => this.downloadReport('csv')}>
                                        Export as CSV
                                    </DropdownItem>
                                ]}
                            />
                        </ToolbarItem>
                    </ToolbarGroup>
                </InventoryCmp>
            </React.Fragment>
        );
    }
}

function mapStateToProps(state, ownProps) {
    return {
        affectedSystems: sExposedSystemsTable(state, ownProps.cve),
        statusList: state.VulnerabilitiesStore.statusList,
        entities: state.entities
    };
}

const mapDispatchToProps = () => {
    return {
        fetchData: (cve, options) => dispatchAction(fetchAffectedSystemsByCVE(cve, options)),
        fetchStatusList: () => dispatchAction(fetchSystemCveStatusList())
    };
};

export default withRouter(
    connect(
        mapStateToProps,
        mapDispatchToProps
    )(SystemsExposedTable)
);
