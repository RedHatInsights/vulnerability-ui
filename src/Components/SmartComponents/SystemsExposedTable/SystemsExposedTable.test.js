import SystemsExposedTable from './SystemsExposedTable';
import { mountWithIntl } from '../../../Helpers/MiscHelper';
import { BrowserRouter as Router } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import { initialState } from '../../../Store/Reducers/CVEDetailsPageStore'
import { Provider, useSelector } from 'react-redux';
import configureStore from 'redux-mock-store';
import React from 'react';
import { act } from "react-dom/test-utils";
import DownloadReport from '../../../Helpers/DownloadReport';

jest.spyOn(React, 'createRef').mockReturnValueOnce({ current: false });
jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn()
}));

jest.mock("../../../Helpers/DownloadReport", () => ({
    ...jest.requireActual("../../../Helpers/DownloadReport"),
    exec: jest.fn()
}));

const state = {
    ...initialState,
    affectedSystemsByCVE: {
        isLoading: false,
        expandedRows: [],
        payload: {
            meta: {},
            data: [
                {
                    id: "c3f36a42-5289-4940-8dcf-d1b7c8f90db2",
                    attributes: {
                        cve_status_id: 2,
                        display_name: "f375c7a8-c680-44fc-9fc4-003de8896877",
                        inventory_id: "f375c7a8-c680-44fc-9fc4-003de8896877",
                        last_evaluation: "2020-07-21T08:14:11.877214+00:00",
                        cve_status_id: 2,
                        reporter: 1,
                        rule: null,
                        rules_evaluation: "2020-07-21T08:14:13.272253+00:00",
                        status_id: 2,
                        status_text: "testhello",
                        status_name: "On-Hold"
                    }
                }
            ]
        }
    }
};

const customMiddleWare = store => next => action => {
    useSelector.mockImplementation(callback => {
        return callback({ CVEDetailsPageStore: state });
    });
    React.createRef.mockImplementation(() => ({ current: { onRefreshData: () => { } } }));
    next(action);
}
const mockStore = configureStore([customMiddleWare]);
let store = mockStore(initialState);

window.insights = {
    loadInventory: jest.fn(() => Promise.resolve({
        inventoryConnector: (args) => {
            const InventoryTable = () => <div>A mock passed!</div>;
            return ({ InventoryTable });
        },
        mergeWithEntities: () => { },
        mergeWithDetail: () => { }
    }))
};

let wrapper;
beforeEach(() => {
    store.clearActions();
    useSelector.mockImplementation(callback => {
        return callback({ CVEDetailsPageStore: initialState });
    });

    wrapper = mountWithIntl(
        <Provider store={store}>
            <Router>
                <SystemsExposedTable
                    cveName={'testCveName'}
                    filterRuleValues={[{}]}
                    cveStatusDetails={{
                        id: 'testCveName',
                        status_id: 1,
                        justification: 'testJustification'
                    }}
                    methods={{ showStatusModal: jest.fn() }}
                />
            </Router>
        </Provider>
    );
});

afterEach(() => {
    useSelector.mockClear();
});

describe('SystemsExposedPage', () => {
    it('Should clear store on unmount', () => {
        wrapper.unmount();
        const dispatchedActions = store.getActions();
        setTimeout(() => {
            expect(dispatchedActions.filter(item => item.type === 'CLEAR_CVE_STORE')).toHaveLength(1);
            expect(dispatchedActions.filter(item => item.type === 'CLEAR_INVENTORY_STORE')).toHaveLength(1);
        })
    });

    it('Should dispatch FETCH_AFFECTED_SYSTEMS_BY_CVE only once on load', () => {
        const dispatchedActions = store.getActions();
        expect(dispatchedActions.filter(item => item.type === 'FETCH_AFFECTED_SYSTEMS_BY_CVE')).toHaveLength(1);
    });

    it('Should update inventory', () => {
        const testObj = { page: 2, per_page: 2 }
        wrapper.update();
        act(() => wrapper.find('BaseInvTable').props().onRefresh(testObj));
        const dispatchedActions = store.getActions();
        expect(dispatchedActions[2]).toEqual({
            type: 'CHANGE_EXPOSED_SYSTEMS_PARAMETERS',
            payload: { page: 2, page_size: 2 }
        });
    });
});