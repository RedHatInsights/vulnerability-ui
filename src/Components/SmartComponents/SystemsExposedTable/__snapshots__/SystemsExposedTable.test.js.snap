// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemsExposedPage Should render without props 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <injectIntl(withRouter(SystemsExposedTable))
        cve="testCveName"
        cveStatusDetails={
          Object {
            "id": "testCveName",
            "justification": "testJustification",
            "status_id": 1,
          }
        }
        filterRuleValues={
          Array [
            Object {},
          ]
        }
        methods={
          Object {
            "showStatusModal": [MockFunction],
          }
        }
      >
        <withRouter(SystemsExposedTable)
          cve="testCveName"
          cveStatusDetails={
            Object {
              "id": "testCveName",
              "justification": "testJustification",
              "status_id": 1,
            }
          }
          filterRuleValues={
            Array [
              Object {},
            ]
          }
          intl={
            Object {
              "defaultFormats": Object {},
              "defaultLocale": "en",
              "formatDate": [Function],
              "formatDateToParts": [Function],
              "formatDisplayName": [Function],
              "formatList": [Function],
              "formatMessage": [Function],
              "formatNumber": [Function],
              "formatNumberToParts": [Function],
              "formatPlural": [Function],
              "formatRelativeTime": [Function],
              "formatTime": [Function],
              "formatTimeToParts": [Function],
              "formats": Object {},
              "formatters": Object {
                "getDateTimeFormat": [Function],
                "getDisplayNames": [Function],
                "getListFormat": [Function],
                "getMessageFormat": [Function],
                "getNumberFormat": [Function],
                "getPluralRules": [Function],
                "getRelativeTimeFormat": [Function],
              },
              "locale": "en",
              "messages": Object {
                "default.cancel": "Cancel",
                "default.delete": "Delete",
                "default.remove": "Remove",
                "default.save": "Save",
              },
              "onError": [Function],
              "textComponent": Symbol(react.fragment),
              "timeZone": undefined,
              "wrapRichTextChunksInFragment": undefined,
            }
          }
          methods={
            Object {
              "showStatusModal": [MockFunction],
            }
          }
        >
          <SystemsExposedTable
            cve="testCveName"
            cveStatusDetails={
              Object {
                "id": "testCveName",
                "justification": "testJustification",
                "status_id": 1,
              }
            }
            filterRuleValues={
              Array [
                Object {},
              ]
            }
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            intl={
              Object {
                "defaultFormats": Object {},
                "defaultLocale": "en",
                "formatDate": [Function],
                "formatDateToParts": [Function],
                "formatDisplayName": [Function],
                "formatList": [Function],
                "formatMessage": [Function],
                "formatNumber": [Function],
                "formatNumberToParts": [Function],
                "formatPlural": [Function],
                "formatRelativeTime": [Function],
                "formatTime": [Function],
                "formatTimeToParts": [Function],
                "formats": Object {},
                "formatters": Object {
                  "getDateTimeFormat": [Function],
                  "getDisplayNames": [Function],
                  "getListFormat": [Function],
                  "getMessageFormat": [Function],
                  "getNumberFormat": [Function],
                  "getPluralRules": [Function],
                  "getRelativeTimeFormat": [Function],
                },
                "locale": "en",
                "messages": Object {
                  "default.cancel": "Cancel",
                  "default.delete": "Delete",
                  "default.remove": "Remove",
                  "default.save": "Save",
                },
                "onError": [Function],
                "textComponent": Symbol(react.fragment),
                "timeZone": undefined,
                "wrapRichTextChunksInFragment": undefined,
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            methods={
              Object {
                "showStatusModal": [MockFunction],
              }
            }
          >
            <Component
              hasGutter={true}
            >
              <div
                className="pf-l-stack pf-m-gutter"
              >
                <Component>
                  <div
                    className="pf-l-stack__item"
                  >
                    <Component>
                      <div
                        className="pf-c-content"
                      >
                        <Component
                          component="h2"
                        >
                          <h2
                            className=""
                            data-pf-content={true}
                          >
                            Exposed systems
                          </h2>
                        </Component>
                      </div>
                    </Component>
                  </div>
                </Component>
                <Component>
                  <div
                    className="pf-l-stack__item"
                  >
                    <Component
                      actionsConfig={
                        Object {
                          "actions": Array [
                            "",
                            Object {
                              "label": "Edit status",
                              "onClick": [Function],
                              "props": Object {
                                "isDisabled": true,
                              },
                            },
                          ],
                          "kebabToggleProps": Object {
                            "isDisabled": true,
                          },
                        }
                      }
                      activeFiltersConfig={
                        Object {
                          "filters": Array [],
                          "onDelete": [Function],
                        }
                      }
                      bulkSelect={
                        Object {
                          "checked": false,
                          "count": undefined,
                          "isDisabled": false,
                          "items": Array [
                            Object {
                              "onClick": [Function],
                              "title": "Select none (0 items)",
                            },
                            Object {
                              "onClick": [Function],
                              "title": "Select page (1 item)",
                            },
                            Object {
                              "onClick": [Function],
                              "title": "Select all (0 items)",
                            },
                          ],
                          "onSelect": [Function],
                        }
                      }
                      dedicatedAction={
                        <Memo(Connect(withRouter(Connect(RouterParams))))
                          cves="testCveName"
                        />
                      }
                      expandable="true"
                      exportConfig={
                        Object {
                          "isDisabled": false,
                          "onSelect": [Function],
                        }
                      }
                      filterConfig={
                        Object {
                          "items": Array [
                            Object {
                              "filterValues": Object {
                                "aria-label": "search-field",
                                "id": "search-systems.search.name",
                                "onChange": [Function],
                                "placeholder": "Filter by name",
                                "value": undefined,
                              },
                              "label": "Name",
                              "type": "text",
                            },
                            Object {
                              "filterValues": Object {
                                "items": Array [
                                  Object {
                                    "label": "Has security rule",
                                    "value": "true",
                                  },
                                  Object {
                                    "label": "Does not have security rule",
                                    "value": "false",
                                  },
                                  Object {
                                    "label": undefined,
                                    "value": undefined,
                                  },
                                ],
                                "onChange": [Function],
                                "value": Array [],
                              },
                              "label": "Security rules",
                              "type": "checkbox",
                            },
                            Object {
                              "filterValues": Object {
                                "items": Array [
                                  Object {
                                    "label": "Not reviewed",
                                    "value": "0",
                                  },
                                  Object {
                                    "label": "In review",
                                    "value": "1",
                                  },
                                  Object {
                                    "label": "On-hold",
                                    "value": "2",
                                  },
                                  Object {
                                    "label": "Scheduled for patch",
                                    "value": "3",
                                  },
                                  Object {
                                    "label": "Resolved",
                                    "value": "4",
                                  },
                                  Object {
                                    "label": "No action - risk accepted",
                                    "value": "5",
                                  },
                                  Object {
                                    "label": "Resolved via mitigation",
                                    "value": "6",
                                  },
                                ],
                                "onChange": [Function],
                                "value": Array [],
                              },
                              "label": "Status",
                              "type": "checkbox",
                            },
                          ],
                        }
                      }
                      hasCheckbox={true}
                      isLoaded={true}
                      items={
                        Array [
                          Object {
                            "children": <InsightsNoSystemRule />,
                            "cve_status_id": 2,
                            "display_name": "f375c7a8-c680-44fc-9fc4-003de8896877",
                            "id": "c3f36a42-5289-4940-8dcf-d1b7c8f90db2",
                            "inventory_id": "f375c7a8-c680-44fc-9fc4-003de8896877",
                            "isOpen": false,
                            "last_evaluation": "2020-07-21T08:14:11.877214+00:00",
                            "reporter": 1,
                            "rule": null,
                            "rules_evaluation": "2020-07-21T08:14:13.272253+00:00",
                            "status": "On-Hold",
                            "status_id": 2,
                            "status_name": "On-Hold",
                            "status_text": "testhello",
                          },
                        ]
                      }
                      key="inventory"
                      onExpandClick={[Function]}
                      onRefresh={[Function]}
                      page={1}
                      perPage={20}
                      showActions={true}
                      showTags={true}
                      tableProps={
                        Object {
                          "actionResolver": [Function],
                          "canSelectAll": false,
                          "isStickyHeader": true,
                          "onSort": [Function],
                          "sortBy": Object {},
                        }
                      }
                    >
                      <div>
                        Loading...
                      </div>
                    </Component>
                  </div>
                </Component>
              </div>
            </Component>
          </SystemsExposedTable>
        </withRouter(SystemsExposedTable)>
      </injectIntl(withRouter(SystemsExposedTable))>
    </Router>
  </BrowserRouter>
</Provider>
`;
